src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesArray = [];
+  const copyState = { ...state };
```

Comment:
It may be better to create new object in every case and do not mutate copied state. But I am not sure this is a mistake. Everything else looks fine!